File: _REPO/APRESS/distributed-.net-programming-in-csharp/Chapter09/BinaryFormatter/Sender/SenderMain.cs
using System;
using System.Messaging;
using CustomerLibrary;
using System.Runtime.Serialization.Formatters;

namespace Sender
{	
class BinarySenderMain
{	
   static void Main(string[] args)
   {        
      // Create the queue instance
      MessageQueue mq = new MessageQueue(@".\private$\MyPrivateQ");
      
      Message msg = new Message();
      msg.Label = "A Customer object";
      msg.Formatter = new BinaryMessageFormatter();

      do
      {
         // Construct Customer and send to queue
         msg.Body = new Customer("Homer", "hsimpson@atomic.com", "333-33-3333");     
         mq.Send(msg);
      } while(Console.ReadLine() != "q");
   }
}
}

File: _REPO/APRESS/distributed-.net-programming-in-vb-.net/Chapter09/XmlFormatter/Sender/SenderMain.vb
Imports System.Messaging
Imports CustomerLibrary

Module SenderMain

   Sub Main      
      'Create the queue instance
      Dim mq As New MessageQueue(".\private$\MyPrivateQ")

      Dim msg As New Message()

      'Place Customer type name (including assembly details) in message label
      msg.Label = GetType(Customer).AssemblyQualifiedName
    
      Do
	      'Construct Customer and send to queue
	      msg.Body = New Customer("Homer", "hsimpson@atomic.com", "5555")    
	      mq.Send(msg)
	   Loop Until Console.ReadLine()= "q"
	   
   End Sub
    
End Module
